### PR TITLE
Model runs

### PR DESCRIPTION
**Pull request by urtrapped247**
Using joseph.ipynb to try runs based on code in omr.py. DataSet is locally stored - 10k files of single lines of random music.

### CODE REVIEW

File: consonance/ml_logic/omr.py [hunk 1]
```diff
@@ -4,35 +4,69 @@
...
 import keras
 from keras import ops
 from keras import layers
-
+from keras.preprocessing.sequence import pad_sequences
 
 
 # Path to the data directory
-data_dir = Path("/Users/ninjamac/code/OMR-research/data")
+data_dir = Path("/Users/ninjamac/code/DataSets/images_cropped")
```
Comment (emzys) [inline consonance/ml_logic/omr.py:20]: Please use os.path module, we don't all have a ninjamac to work with:)
Example:

`file_path = os.path.join(os.getcwd(), "consonance/api/hot-cross-buns.mid")`

File: consonance/ml_logic/omr.py [hunk 2]
```diff
@@ -4,35 +4,69 @@
...
 
 import tensorflow as tf
 import keras
 from keras import ops
 from keras import layers
-
+from keras.preprocessing.sequence import pad_sequences
 
 
 # Path to the data directory
```
Comment (emzys) [inline consonance/ml_logic/omr.py:19]: It's best if executable code is wrapped in a function in a python code. In the notebook you can run all cells from top to bottom, but it works a bit differently in a web app.

Constants or environment variables can be declared outside of a function scope and can be imported/accessed globally/within a file scope, but executable code which relies on computed variables can break an app build if the values needed are not available in the current state. Make sure you have access to these variables and their values are not empty at all times the code is accessed by the app.

Eg. max_length variable broke the build because labels were not available.
In cases like this it's safer if this code is included inside of a function, and checks are in place to see if our data has the desired type and value.

File: consonance/ml_logic/omr.py [hunk 3]
```diff
@@ -4,35 +4,69 @@
...
-data_dir = Path("/Users/ninjamac/code/OMR-research/data")
+data_dir = Path("/Users/ninjamac/code/DataSets/images_cropped")
+
+# Load the CSV file
+csv_path = "/Users/ninjamac/code/DataSets/music/labels.csv"
```
Comment (emzys) [inline consonance/ml_logic/omr.py:23]: what are the contents of this file? is it always there?

File: consonance/ml_logic/omr.py [hunk 4]
```diff
@@ -4,35 +4,69 @@
...
+    'B3_16th': 68, 'C4_16th': 69, 'D4_16th': 70, 'E4_16th': 71, 'F4_16th': 72, 'G4_16th': 73, 'A4_16th': 74, 'B4_16th': 75, 'C5_16th': 76,
+    'D5_16th': 77, 'E5_16th': 78, 'F5_16th': 79, 'G5_16th': 80, 'A5_16th': 81, 'B5_16th': 82, 'C6_16th': 83, 'D6_16th': 84
+}
 
 # Batch size for training and validation
 batch_size = 16
 
 # Desired image dimensions
-img_width = 500
+img_width = 350
```
Comment (emzys) [inline consonance/ml_logic/omr.py:69]: I assume I need to change this value in preprocessing for the prediction too?

File: consonance/ml_logic/omr.py [hunk 5]
```diff
@@ -314,7 +362,9 @@ def decode_batch_predictions(pred):
     # Iterate over the results and get back the text
```
Comment (emzys) [inline consonance/ml_logic/omr.py:362]: I need to repurpose this for the predictions, but tbh I'm still not positive as to what the max_length and the variables it's computed based on (images and labels) represent exactly. Not sure what file name has to do with prediction. Could you please explain?

**Review by emzys — COMMENTED** [review]
Hey @urtrapped247 , Here are some suggestion as to how to refactor code when you extract it from notebook in the future, as well as some questions regarding the code 🙇😊

File: consonance/ml_logic/omr.py (hunk 3 above)
Comment (urtrapped247) [reply consonance/ml_logic/omr.py:23]: This was code to point to the locations of the images folder and the label file. This is not always there. I would say it is not necessary for production, but if someone else wanted to train their own model off of this code, they would need to define it.

File: consonance/ml_logic/omr.py (hunk 4 above)
Comment (urtrapped247) [reply consonance/ml_logic/omr.py:69]: Any prediction would need to follow the dimensions that the model was trained on, yes. This would be part of preprocessing for any image we run a prediction on.

File: consonance/ml_logic/omr.py (hunk 3 above)
Comment (urtrapped247) [reply consonance/ml_logic/omr.py:23]: the label file is a csv with a layout as follows:
filename (string), labels (list of strings). 

filename,labels
music_0.png,"['A5_half', 'E4_16th', 'C4_eighth', 'C5_eighth', 'D6_eighth', 'A4_16th', 'E4_whole', 'B5_eighth', 'C4_16th', 'D5_half', 'E5_quarter', 'F5_16th', 'D6_quarter', 'C5_quarter', 'G4_eighth', 'B5_16th', 'C4_quarter', 'C5_16th', 'D5_16th', 'B5_half', 'D5_quarter', 'A4_eighth', 'E5_16th']"
